src/v2/providers/alerts/alerts.ts

```diff
@@ -0,0 +1,127 @@
+import { ManifestEndpoint } from '../../../common/manifest';
+import { CloudEvent, CloudFunction } from '../../core';
+import * as options from '../../options';
+
+/** Data */
+export interface FirebaseAlertData<T = any> {
+  createTime: string;
+  endTime: string;
+  payload: T;
+}
+
+/** @internal */
+export const eventType = 'firebase.firebasealerts.alerts.v1.published';
+
+/** The underlying alert type of the Firebase Alerts provider */
+export type AlertType =
+  | 'crashlytics.newFatalIssue'
+  | 'crashlytics.newNonfatalIssue'
+  | 'crashlytics.regression'
+  | 'crashlytics.stabilityDigest'
+  | 'crashlytics.velocity'
+  | 'crashlytics.newAnrIssue'
+  | 'billing.planUpdate'
+  | 'billing.automatedPlanUpdate'
+  | 'appDistribution.newTesterIosDevice'
+  | string;
+
+/** Options */
+export interface FirebaseAlertOptions extends options.EventHandlerOptions {
+  alertType: AlertType;
+  appId?: string;
+}
+
+/** Cloud Event Type */
+interface WithAlertTypeAndApp {
+  alertType: string;
+  appId?: string;
+}
+export type AlertEvent<T> = CloudEvent<
+  FirebaseAlertData<T>,
+  WithAlertTypeAndApp
+>;
+
+/** Handlers */
+/** Handle an alert published */
+export function onAlertPublished<T extends { ['@type']: string } = any>(
+  alertTypeOrOpts: AlertType | FirebaseAlertOptions,
+  handler: (event: AlertEvent<T>) => any | Promise<any>
+): CloudFunction<FirebaseAlertData<T>> {
+  const [opts, alertType, appId] = getOptsAndAlertTypeAndApp(alertTypeOrOpts);
+
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(
+      raw as CloudEvent<FirebaseAlertData<T>, WithAlertTypeAndApp>
+    );
+  };
+
+  func.run = handler;
+
+  // TypeScript doesn't recognize defineProperty as adding a property and complains
+  // that __endpoint doesn't exist. We can either cast to any and lose all type safety
+  // or we can just assign a meaningless value before calling defineProperty.
+  func.__trigger = 'silence the transpiler';
+  func.__endpoint = {} as ManifestEndpoint;
+  defineEndpoint(func, opts, alertType, appId);
+
+  return func;
+}
+
+/**
+ * @internal
+ * Helper function for defining an endpoint for alert handling functions
+ */
+export function defineEndpoint(
+  func: CloudFunction<FirebaseAlertData<any>>,
+  opts: options.EventHandlerOptions,
+  alertType: string,
+  appId?: string
+): void {
+  Object.defineProperty(func, '__endpoint', {
```

Comment:
I don't think this has to be a dynamic property. We did this in storage trigger only because we needed to get `bucket` from `FIREBASE_CONFIG`, but that's not the case here. See `https.ts` for an example of static __endpoint definition!

Reply:
Makes sense, changed to static definition